### ISSUE TITLE
The directory Id generated can be too long

### ISSUE THREAD

**Issue by Technus** [oleg-shilo/wixsharp#1244]
After couple directories in the path the Dir ID becomes too long, breaking reference counting and stuff.

It should default to something like:

```Dir.{Name}(.{Number})```

Example: 
Before - ```CommonAppDataFolder.SomeLongDomainName.SomeLongApplicationName.Settings```
After - ```Dir.Settings```

https://github.com/oleg-shilo/wixsharp/blob/5f8508a27c670bab10432ec8bb01141b5f1d0557/Source/src/WixSharp/Compiler.cs#L3445

**Comment by oleg-shilo** [comment]
Unfortunately, this cannot be a good choice of behaviour as it implies that the path of the folder consists only of unique names. 
The folder structure like this `%prog_files%\poduct\service\dependant_service\dependant_service` will break.

If it is a problem in your case I suggest you assign IDs explicitly in the constructor.

**Comment by Technus** [comment]
...
Dir.dependant_service
Dir.dependant_service.1
...

Not really it's done like that with the default gen for other objects already.

**Comment by oleg-shilo** [comment]
Actually, you are right.

The id-algorithm in question was a legacy routine that was designed a long time ago and the later phase of id-generation simplification accidentally overlooked `Dir` types, while covering all other types.

I have fixed it on [b64ae4e](https://github.com/oleg-shilo/wixsharp/commit/b64ae4ec660af9cd78712949c21b933bfe858128).
The fix will be available in the very next release.

To ensure a minimal impact the old id-generation algorithm is also available via `AutoElements.LegacyDirIdAlgorithm`. 
 
The new behaviour:

```C#
var project =
        new ManagedProject("My Product",
            new Dir(@"%ProgramFiles%\My Product",
                new File("myapp.cs"),
                new Dir("bin",
                    new Dir("help",
                        new Dir("settings",
                            new Dir("icons"),
                            new Dir("settings",
                                new File("myapp.exe")))))));
```

```xml
<Directory Id="bin" Name="bin">
  <Directory Id="help" Name="help">
    <Directory Id="settings" Name="settings">
      <Directory Id="icons" Name="icons">
      </Directory>
      <Directory Id="settings.1" Name="settings">
      </Directory>
    </Directory>
  </Directory>
</Directory>
```

The old behaviour with `AutoElements.LegacyDirIdAlgorithm = true`:

```xml
<Directory Id="INSTALLDIR.bin" Name="bin">
  <Directory Id="INSTALLDIR.bin.help" Name="help">
    <Directory Id="INSTALLDIR.bin.help.settings" Name="settings">
      <Directory Id="INSTALLDIR.bin.help.settings.icons" Name="icons">
      </Directory>
      <Directory Id="INSTALLDIR.bin.help.settings.settings" Name="settings">
      </Directory>
    </Directory>
  </Directory>
</Directory>

```
